### ISSUE TITLE
Ollama reports incorrect version and does not show up in System tray

### ISSUE THREAD

**Issue by VirtualZardoz** [ollama/ollama#4944]
### What is the issue?

Ever since I upgraded Ollama to version 0.1.38, it continues reporting this as its version number:
```
[...]> ollama -v
ollama version is 0.1.38
```
Despite the fact that I have updated it to all the released versions since 0.1.38. In fact my current version should be reported as 0.1.42.

Also, Ollama has stopped showing up in the System tray. Which seems to limit some controls I have over the application.

I have done the usual before posting here:
- Searched the Discord
- Searched this Github
- Uninstalled Ollama
- Reinstalled Ollama

Ollama has been installed with the package downloaded from the website. I run on Windows 11, Nvidia Geforce RTX 3090.

### OS

Windows

### GPU

Nvidia

### CPU

AMD

### Ollama version

0.1.42

**Comment by VirtualZardoz** [comment]
Thanks a lot for your comment. Here is where I stand:

> * Have you taken a look into the logs at _C:\Users\[username]\AppData\Local\Ollama_? There might be error messages indicating the roots of the problems

Sharing my logs here, I cannot see what's wrong, except maybe the mention of the other instance of Ollama running, from god knows where ?
[app.log](https://github.com/user-attachments/files/15776344/app.log)



> * System tray: check in the Task Manager the "Autostart" tab. Is the program _ollama.exe_ deactivated there?

The Autostart is ENABLED.

> * Alternatively to the desktop app you can run `ollama serve`

Yes, that works. Ollama actually works on my PC. Just the absence of sys tray icon and the incorrect version being reported.


> * Use Revo Uninstaller to uninstall ollama cleanly and reinstall it again.

I did that before but no success.

**Comment by d-kleine** [comment]
```
time=2024-06-09T08:16:23.897+02:00 level=INFO source=lifecycle.go:70 msg="Detected another instance of ollama running, exiting"
time=2024-06-09T08:18:46.711+02:00 level=INFO source=logging.go:47 msg="ollama app started"
```

As long you did not run `ollama serve`, this looks like there is a second instance of ollama running on your system.

I just tested that, I have  `ollama serve` in a terminal running in the background, the Ollama desktop app tries to launch, but fails and the sys tray icon disappears quickly, like you described the problem.

For me, it looks like this is a problem on your side as you have two ollama instances running at the same time, maybe due to duplicated installs of ollama.

Have you tried uninstalling ollama with Revo Uninstaller and checking it stills - after uninstalling, without reinstalling - shows a version with `ollama -v`?

**Comment by VirtualZardoz** [comment]
> I just tested that, I have `ollama serve` in a terminal running in the background, the Ollama desktop app tries to launch, but fails and the sys tray icon disappears quickly, like you described the problem.

Yes, it indeed seems that there are two instances running and that they cause the problem. However, I did not invoke two instances, so this is strange.

> Have you tried uninstalling ollama with Revo Uninstaller and checking it still - after uninstalling, without reinstalling - shows a version with `ollama -v`?

I uninstalled ollama with Revo Uninstaller Pro. Here is the output of the `ollama -v` command after uninstall is complete:
![image](https://github.com/ollama/ollama/assets/167669409/ba95086a-47f8-4ce5-b718-90ca5d5d5d70)

I then reinstalled Ollama by downloading a fresh copy of the installer from Ollama website. Here is the output of the `ollama -v` command after reinstalling:
![image](https://github.com/ollama/ollama/assets/167669409/ee8a5334-56ca-48a1-96d4-5c72e5be28fd)
So there is an error in reporting the correct version number.

Also, this is what I get when I immediately after run the `ollama serve` command:
![Screenshot 2024-06-11 095327](https://github.com/ollama/ollama/assets/167669409/900de608-7bab-41a1-9c40-8abe255121a2)


So Ollama generates another instance of itself, without me asking it to do that. 

Here is the new logs 
[app.log](https://github.com/user-attachments/files/15784250/app.log)
And of course, Ollama is not in sys tray.

It is probably something on my end though, but just sharing in case someone knows of a fix. Thanks.

**Comment by VirtualZardoz** [comment]
> Are you logged in as admin on your pc?

No. And I install Ollama as the regular user and it does not prompt me to give any admin credentials during install process.

**Comment by d-kleine** [comment]
One last idea:
See log: `time=2024-06-11T09:45:50.305+02:00 level=INFO source=store.go:96 msg="wrote store: C:\\Users\\Shahram\\AppData\\Local\\Ollama\\config.json"`
Uninstall Ollama und delete the folder `C:\Users\Shahram\AppData\Local\Ollama`, maybe there are still leftovers.
Also remove the environment variables, see here https://github.com/ollama/ollama/blob/main/docs/faq.md#setting-environment-variables-on-windows

**Comment by VirtualZardoz** [comment]
> One last idea: 

I think that thanks to your suggestions, I now have isolated the problem. As soon as I install Ollama and run it for the first time, something else on my computer also initiates an instance of Ollama server and the desktop app quits.

Indded, after I uninstall with Revo and wipe every bit of Ollama leftovers, as you suggested, the `ollama` command does nothing. But as soon as I reinstall, I have a call from somewhere that starts an Ollama server automatically and starts conflicting with the desktop app.

```
PS C:\Users\Shahram> ollama -v
ollama version is 0.1.38
Warning: client version is 0.1.42
```
I thought that it may be one of these containers, and I stopped all of them while reinstalling and doing all the testing. But even with these guys stopped, and checked that they remain stopped after a restart, the problem persisted.
![image](https://github.com/ollama/ollama/assets/167669409/c93de895-b150-49d8-844c-a8882a0f9e2d)

**Comment by d-kleine** [comment]
> I think that thanks to your suggestions, I now have isolated the problem. As soon as I install Ollama and run it for the first time, something else on my computer also initiates an instance of Ollama server and the desktop app quits.

Yeah, that is what I am saying.

Final thoughts on what you could try to find the cause of this problem:
* Use the Task Manager, clicking through the tabs if you might see duplicated instances and their roots (there is also a "app history" tab, maybe there is something helpful inside)
* Creating a dump file and read it via WinDbg
* Checking the Event Viewer when installing to what happens when installing

Good luck 🍀

**Comment by VirtualZardoz** [comment]
@d-kleine 

I found the culprit which starts an instance of Ollama and it is the Docker engine despite the fact that all its containers, especially those working with Ollama, were stopped all the time.

Also, the Ollama instance being started by Docker does not appear anywhere in Task or Process list. At least not under the name Ollama. 

So I guess that the new area of research for troubleshooting should now move to analyzing why Docker does that and how to prevent it. I will do that now.

Again, thanks a lot for your incredible support. I truly appreciate.

**Comment by VirtualZardoz** [comment]
I am unfortunately reopening this issue.

The behavior of multiple instances of Ollama launching and conflicting is persisting.

From previous feedback, I have now removed (deleted) all Docker containers which have a dependency on Ollama. I have completely uninstalled Ollama with Revo Uninstaller. 

I have reinstalled a clean copy of Ollama.

I have removed the Ollama shortcut from `C:\Users\Shahram\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup`

So I expect Ollama not to start at all when I reboot the computer. But it DOES and does it multiple times, as can be seen from this log:

```
time=2024-06-16T08:41:51.521+02:00 level=INFO source=logging.go:47 msg="ollama app started"
time=2024-06-16T08:41:51.540+02:00 level=INFO source=store.go:96 msg="wrote store: C:\\Users\\Shahram\\AppData\\Local\\Ollama\\config.json"
time=2024-06-16T08:41:51.553+02:00 level=INFO source=store.go:96 msg="wrote store: C:\\Users\\Shahram\\AppData\\Local\\Ollama\\config.json"
time=2024-06-16T08:41:51.556+02:00 level=INFO source=lifecycle.go:70 msg="Detected another instance of ollama running, exiting"
```
More confusing, I cannot find any process named ollama:
![image](https://github.com/ollama/ollama/assets/167669409/e1eb9ec1-5d28-42b3-92a6-194623d724f3)
And yet, it is running:
![image](https://github.com/ollama/ollama/assets/167669409/d1a1dbb7-2ea5-4711-b183-232691cea8bd)
